test/alternator/run

```diff
@@ -24,6 +24,15 @@ urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 print('Scylla is: ' + run.find_scylla() + '.')
 
 extra_scylla_options = []
+remove_scylla_options = []
+
+# If the "--vnodes" option is given, drop the "tablets" experimental
+# feature (turned on in run.py) so that all tests will be run with the
+# old vnode-based replication instead of tablets. This option only has
+# temporary usefulness, and should eventually be removed.
+if '--vnodes' in sys.argv:
+    sys.argv.remove('--vnodes')
+    remove_scylla_options.append('--experimental-features=tablets')
```

Comment:
Doesn't this mean that we'll default to not testing alterantor streams and ttl any more?

Reply:
Yes, it sort of does, because indeed these two things no longer work with tablets.
We have the same issue about no longer testing CDC in cql-pytest (since @denesb 's patch for cql-pytest got in)

There are two "solutions" to this observation, if it needs to be solved:
1. Have CI run each tests twice (not every developer needs to do this, it will be very annoying).
2. The test code could, instead of "xfailing" a CDC test in tablets mode, use a different tablet-less keyspace for the CDC tests. It's not very hard to do, actually. The downside is that it won't be visible that these tests don't work in tablets - they can be in this state that they only work on vnodes for years, and nobody will notice (developers won't see a lot of xfails or skips when running these tests).

Any opinion on what we should do here?

Reply:
Let's revert that patch then.

Reply:
I don't see any reason to revert the patch - it doesn't break anything in the code (the worst it can break is not to test something). We can do a followup patch to fix the cql-pytest situation in anyway we want CC @denesb .

I'll see what I can do for **this** patch (Alternator tests) to make you happier. I can try option 2 - CDC and TTL will be temporarily tested without tablets, just so we keep testing them somehow and not just skip them.

But I think option 1 - running all the tests of all types twice, once for vnodes and once for tablets - from the CI, is the only option that will make you feel 100% safe, ensuring that a year from now, we continue to test (even if just once a week, not on every CI run) that all the tests pass even if vnodes are used.

Reply:
It breaks our ability to detect regressions. IMO that's critical.

Reply:
I'm also against reverting Botond's patch. Temporarily not testing something is not at the same severity as temporarily breaking a feature, and as Botond said, while it's temporarily not testing something (e.g., CDC) is is testing something which is more important and more urgent (tablets) so it's a good compromise, until a follow up patch that does both.

I'm working now on a patch for Alternator (to replace what I have right now in this PR) which will 1. run all the tests with tablets, except 2. specific tests that don't work with tablets will temporarily be run without tablets.
In the future we can, and should, consider CI to run every test (including test.py and dtest) twice, once with tablets and once with vnodes. But it's not a blocker, we can do it later.

Reply:
I will send a PR to run CDC tests with vnodes, until they are made to work with tablets. After which points, they should be ran with both, because they apparently do care about the replication method.

Reply:
"Temporarily" means nothing without a date.

Reply:
@avikivity 
1. Definitely, a tablets-specific fix should add a tablets-specific test. We've been doing this all along - a lot of these tests are in test/topology_*/*tablets*, and there's even one in test/cql-pytest. But Botond's CQL patch and my Alternator patch is not about those tests, it's about the almost 2000 other single-node functional tests for CQL and Alternator that 99% of them are completely agnostics to tablets vs vnodes.
2. Alternator TTL, for example, can probably be fixed fairly easily to work on both tablets and  vnodes. Until then, this PR sets the test to run on vnodes instead of tablets. It is "temporary" until the time we do the Alternator TTL fix. If you want a date for this specific fix, I can commit to one, but I got the impression that Dor didn't want to rush Alternator fixes. I just wanted to get the minimum in now.

To be honest, it's not clear to me what you're asking to do. Botond sent in the past a patch which runs every single cql-pytest twice, once with tablets and once without, and you agreed that it's excessive. Now he sent a patch to skip tests for features not yet available for tablets (e.g., CDC), and you didn't like that either. Now I sent a this PR to (in the context of Alternator) run all the tests with tablets and just the missing features (e.g., CDC) will be run but without tablets - you didn't like it either. So, what would you like to do?

Reply:
> "Temporarily" means nothing without a date.

I will send a PR re-enabling all disabled tests today.
Part of it is already here: https://github.com/scylladb/scylladb/pull/16802.